src/main/java/org/apache/commons/lang3/time/StopWatch.java

```diff
@@ -671,4 +686,622 @@ public void unsplit() {
         splitState = SplitState.UNSPLIT;
     }
 
+    /**
+     * Take the time of the exectution of a given {@link FailableSupplier}.
+     *
+     * <p>
+     * <em>Take the time of the execution of given {@link FailableSupplier}</em>
+     * </p>
+     * <pre>{@code
+     * final StopWatch watch = StopWatch.create();
+     * String result = watch.get(() -> "A");
+     * }</pre>
+     *
+     * <p>
+     *  <em>Note:</em>
+     * </p>
+     * <ul>
+     *     <li>Be careful if you pass a {@link FailableSupplier} to other methods, as the wrapped instance depends on
+     *         the state of used {@linkplain StopWatch}</li>
+     * </ul>
+     *
+     * @param <T> the type of return value of the supplier
+     * @param <E> the kind of thrown exception or error
+     * @param supplier the {@link FailableSupplier} those execution should be measured
+     * @return the result of the {@link FailableSupplier#get()} operation
+     * @throws E if the supplier fails
+     * @since 3.18.0
```

Comment:
Hello @obfischer 

This is missing a Javadoc throws tag for IllegalArgumentException like the other methods.

Since this method calls `resumeOrStartStopWatch` like all the others, we can assume:
```Java
@throws IllegalStateException if the {@link StopWatch} is not stopped or suspended
```
BUT, either the Javadoc is wrong (in all the other methods) or the implementation is wrong. According to the Javadoc, this should pass:
```java
        @Test
        public void testGetStopped() {
            final StopWatch watch = StopWatch.create();
            watch.start();
            watch.stop();
            assertTrue(watch.isStopped());
            Supplier<String> supplier = () -> "Foobar";
            assertEquals("Foobar", watch.get(supplier));
            assertTrue(watch.isSuspended());
        }
```
This test throws `IllegalStateException` on the `get()` even though the stopwatch is stopped.

What should happen in this case?

Reply:
Good point, this is a bug. I have to think about it.

Reply:
I would say, that the method `resumeOrStartStopWatch` should

* start the stop watch only if `runningState == State.UNSTARTED`
* resume the stop watch if `runningState.isSuspended() == true`
* every other state should lead to an `IllegalStateException`

Futhermore the JavaDoc should:

* state the behavior above 
* state that the stop watch will be suspend after taking time

WDYT?

Reply:
Ping @garydgregory